src/hotspot/share/opto/cfgnode.cpp

```diff
@@ -2303,11 +2303,9 @@ Node *PhiNode::Ideal(PhaseGVN *phase, bool can_reshape) {
           mms.set_memory(phase->transform(phi));
         }
         if (igvn) { // Unhook.
-          igvn->hash_delete(hook);
-          for (uint i = 1; i < hook->req(); i++) {
-            hook->set_req(i, NULL);
-          }
+          igvn->_worklist.remove(hook);
         }
+        hook->destruct();
```

Comment:
I think we should pass PhaseGVN* into destruct() and do removal from worklist there because it looks like repetitive pattern. Also we can take Compile pointer from PhaseGVN instead of calling Compile::current():
https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/node.cpp#L584

Reply:
Thanks Vladimir, that's a good suggestion. I've updated the PR accordingly.